bin/test.sh

```diff
@@ -51,6 +51,9 @@ write-assoc-testdata() {
 domain,flag..HTTPS
 google.com,1
 google.com,1
+google.com,1
```

Comment:
Why add these lines?  I think it makes it more confusing to interpret.  Right now it's 2/7,2/7, 3/7.  Maybe change it to 10 values, like:

30% google true
20% google false
20% yahoo true
10% yahoo false
10% bing true
10% bing false

That way it's easier to eyeball.  It's hard to do the arithmetic for /7 or /9.